### PR TITLE
gpinitsystem: Use initdb to determine locale

### PR DESCRIPTION
**Pull request by jmcatamney**
There is already logic in initdb to determine the system locale and
initialize the database with the correct locale values, so this commit
removes the existing default locale logic in gpinitsystem in favor of
letting initdb set the locale on the coordinator and then retrieving the
resulting values to use when initializing the segments.

This commit only changes gpinitsystem's behavior with respect to the
default locale, and then only on systems whose locale did not already
match the default.  Specifying one or more locales with the appropriate
flags still has the same behavior.

(cherry picked from commit 267d52c6bdf0893d69cb476e1160688241032828)

Pipeline is [here](https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpinitsystem_locale).

### CODE REVIEW

File: gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py [hunk 1]
```diff
@@ -3113,3 +3113,78 @@ def impl(context, including):
...
+
+@then('content {content} is {desired_state}')
+def impl(context, content, desired_state):
+    acceptable_states = ["balanced", "unbalanced"]
+    if desired_state not in acceptable_states:
+        raise Exception("expected desired state to be one of %s", acceptable_states)
+
+    role_operator = "=" if desired_state == "balanced" else "<>"
+    with dbconn.connect(dbconn.DbURL(dbname="template1"), unsetSearchPath=False) as conn:
+        rows = dbconn.query(conn, "SELECT role, preferred_role FROM gp_segment_configuration WHERE content = %s and preferred_role %s role" % (content, role_operator)).fetchall()
```
Comment (kalensk) [inline gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py:3125]: I believe `query` becomes `execSQL`.

Comment (dkrieger-pvtl) [reply gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py]: fixed in two places. Also, `rows` is a list in 6X, not a dict, so I modified that as well.

File: gpMgmt/bin/gpinitsystem [hunk 2]
```diff
@@ -1212,7 +1254,9 @@ CREATE_QD_DB () {
 		cmd="$INITDB"
 		cmd="$cmd -E $ENCODING"
 		cmd="$cmd -D $GP_DIR"
-		cmd="$cmd --locale=$LOCALE_SETTING"
+		if [ x"" == x"$LOCALE_SETTING" ]; then
```
Comment (asimrp) [inline gpMgmt/bin/gpinitsystem:1257]: Is the condition wrong, should it be `x"" != x"$LOCALE_SETTING"`?

Comment (dkrieger-pvtl) [reply gpMgmt/bin/gpinitsystem:1257]: Good catch...it's also currently merged onto 7X.  

As the code currently is, we actually [set each LC_ individually](https://github.com/jmcatamney/gpdb/blob/gpinitsystem_locale_6X/gpMgmt/bin/gpinitsystem#L524) and, if the `LOCALE_SETTING` is empty, we pass `--locale=`, which in a test initdb seems to just ignore.

But we should fix this...likely by removing these three lines as we do pass `LC_ALL_SETTING` which does take, in order, the gpinitsystem `--lc*` flags, then the `LC_*` environment variables, and then the `--locale` input to gpinitsystem.

@jmcatamney FYI

Comment (dkrieger-pvtl) [reply gpMgmt/bin/gpinitsystem]: FIxed.

**Review by divyeshddv — APPROVED** [review]

**Comment by kalensk** [comment]
This change affects the default locale and since 6X is a stable branch there have been discussions with @ashwinstar and @greenplum-db/cluster-management about doing a minimal change as originally scoped in PR https://github.com/greenplum-db/gpdb/pull/11398.

Specifically, this PR will enable initdb to choose a default locale if the user does not pass one via a flag to gpinitsystem or set an environment variable. The default locale for intitdb is system dependent, but is most likely not utf-8 and probably C. The current 6X behavior of gpintisystem is to default to utf-8. A more conservative approach that retains this default behavior is introduced in https://github.com/greenplum-db/gpdb/pull/11547.

**Pull request closed without merging**